File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/LoginCommandModule.java
package ro.razvan.java.dagger.atm;

import dagger.Binds;
import dagger.BindsOptionalOf;
import dagger.Module;
import dagger.multibindings.IntoMap;
import dagger.multibindings.StringKey;

@Module
public interface LoginCommandModule {

    @Binds
    @IntoMap
    @StringKey(LoginCommand.COMMAND_KEY_LOGIN)
    Command bindLoginCommand(LoginCommand command);

    @BindsOptionalOf
    Database.Account optionalAccount();

}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/LoginCommandModule.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Binds
import dagger.BindsOptionalOf
import dagger.Module
import dagger.multibindings.IntoMap
import dagger.multibindings.StringKey

@Module
interface LoginCommandModule {

    @Binds
    @IntoMap
    @StringKey(COMMAND_KEY_LOGIN)
    fun bindLoginCommand(command: LoginCommand): Command

    @BindsOptionalOf
    fun optionalAccount(): Database.Account

}